eras/shelley/impl/testlib/Test/Cardano/Ledger/Shelley/Imp/UtxowSpec.hs

```diff
@@ -49,3 +56,52 @@ spec = describe "UTXOW" $ do
             mkBasicTx txBody
               & (witsTxL %~ (bootAddrTxWitsL .~ [aliceBadWitness]))
       submitFailingTx txBad [injectFailure $ InvalidWitnessesUTXOW [aliceVKey]]
+
+  it "MissingVKeyWitnessesUTXOW" $ do
+    aliceKh <- impRootKh
+    tx <- do
+      bobAddr <- snd <$> freshKeyAddr
+      pure $
+        mkBasicTx mkBasicTxBody
+          & bodyTxL . outputsTxBodyL .~ [mkBasicTxOut bobAddr (inject (Coin 10))]
+    void $
+      withFixup (addRootTxIn >=> fixupFees) $
```

Comment:
Redundant `void` and relying on root TxIn is a bit convoluted. Why not create a new txIn and use a postfixup? I think something like this should work:
```suggestion
    aliceKh <- freshKeyHash
    txIn <- sendCoinTo (Addr Testnet (KeyHashObj aliceKh) StakeRefNull) 99
    withPostFixup (\tx -> tx & inputsTxBodyL . bodyTxL <>~ [txIn]) $
```